### ISSUE TITLE
IIS Rewrites not working for multiple cultures in current Cloud infrastructure.

### ISSUE THREAD

**Issue by sergiutesu** [umbraco/Umbraco.Cloud.Issues#536]
### Issue description

Issue:
Rewrites work only for the default language, they won't work for another language.

I used CMS Version: 8.17.2

Steps to reproduce
1. Make sure Umbraco has two languages
![image](https://user-images.githubusercontent.com/83580347/152378674-95033eb8-ec91-4628-8aa7-4890c5a7222b.png)

2. Make a doctype which is a culture variant and allow a root node
![image](https://user-images.githubusercontent.com/83580347/152378775-3858de9d-72ac-4074-90b0-c614cc219254.png)

3. Create a node with this doctype for both languages and attach different culture hostnames for each language
![image](https://user-images.githubusercontent.com/83580347/152383601-45791f06-9591-4717-9c03-fbbb9f9e6502.png)
![image](https://user-images.githubusercontent.com/83580347/152383639-88568249-e668-4717-a16f-755c866409c9.png)

4. Write two similar rewrite rules one for each hostname and add them to the web.config

<rule name="EnglishProductsRewrites">
  <match url="(.*?)/c-([0-9]+)/c-([0-9]+)" />
  <conditions>
    <add input="{HTTP_HOST}" pattern="custom-hostname.com" />
    <add input="{REQUEST_URI}" negate="true" pattern="azurewebsites.net" />
  </conditions>
  <action type="Rewrite" url="english-products/?catalog={R:2}" />
</rule>
<rule name="DanishProductRewrites">
  <match url="(.*?)/c-([0-9]+)/c-([0-9]+)" />
  <conditions>
    <add input="{HTTP_HOST}" pattern="custom-hostname.cdk" />
    <add input="{REQUEST_URI}" negate="true" pattern="azurewebsites.net" />
  </conditions>
  <action type="Rewrite" url="dansk-produkter/?catalog={R:2}" />
</rule>

After the above steps are done:
You will notice that the rewrites only work for the default first language and the other languages won't work.

These are some samples:
Trigger the 1st rewrite condition
English version: https://custom-hostname.com.com/english-products/category/c-55/c-66 it's working fine

![image](https://user-images.githubusercontent.com/83580347/152383296-0e708af2-3312-4952-8af4-d7c89f2e2bfd.png)

 
Trigger the 2nd rewrite condition
Danish version: https://custom-hostname.dk/dansk-produkter/category/c-55/c-66 it's giving a 404 template not found.

![image](https://user-images.githubusercontent.com/83580347/152383333-700df801-5a7b-4dc7-a8ee-7a309c22a009.png)

What happens is that in the background it will look for the english-products instead of the dansk-produkter, and it won't be able to find that.

**Comment by ronaldbarendse** [comment]
To give this some context: the new Umbraco Cloud infrastructure adds Cloudflare in front of the website (as reverse proxy, to add CDN/caching capabilities and transparently changing the origin). The `HTTP_HOST` header is automatically replaced with the original hostname, so Umbraco will see the correct hostname and this will work as expected when [setting up Cultures and Hostnames](https://our.umbraco.com/documentation/umbraco-cloud/Set-up/Manage-Hostnames/).

If you rewrite the incoming request using IIS rewrite rules and `<action type="Rewrite" />` however, a new internal request is done within IIS and because that's not preserving to original hostname, that request will see the origin hostname instead. This is basically [setting up a second reverse proxy](https://docs.microsoft.com/en-us/iis/extensions/url-rewrite-module/reverse-proxy-with-url-rewrite-v2-and-application-request-routing).

Maybe @sitereactor can give a definitive answer whether this is officially (or even technically) supported. In any case, I would advice to use redirects instead (by changing `action="Rewrite"` to `action="Redirect"` and optionally add `redirectType="Permanent"`) or configure the hostnames within Umbraco and [write content finders to match the URLs](https://our.umbraco.com/documentation/reference/routing/request-pipeline/icontentfinder).

**Comment by sitereactor** [comment]
@ronaldbarendse What is your source for this?
> If you rewrite the incoming request using IIS rewrite rules and <action type="Rewrite" /> however, a new internal request is done within IIS and because that's not preserving to original hostname, that request will see the origin hostname instead

The link you provided includes ARR and actively enabling Proxy, which on azure webapps, is configured explicitly, as described here https://ruslany.net/2014/05/using-azure-web-site-as-a-reverse-proxy/

**Comment by ronaldbarendse** [comment]
Right, so ARR is installed on Azure Web Apps, but not enabled by default, so it's not possible to configure a second reverse proxy indeed, thanks for clearing that up! 👌🏻 But still, the IIS rewrite _does_ change the incoming request and thereby probably loses the preserved original hostname and uses the origin hostname instead.

Going from [this documentation](https://docs.microsoft.com/en-us/iis/extensions/url-rewrite-module/iis-url-rewriting-and-aspnet-routing), it states _"The URL rewriting module runs early in the request-processing pipeline, modifying the requested URL before the Web server decides which handler to use to process the request."_, so maybe the customer rewrite runs before the original hostname is preserved?

**Comment by jemayn** [comment]
Thank you for reporting this. We're closing this issue as part of a cleanup of older items on the tracker. If you're still experiencing this problem on a current version of Umbraco Cloud, please open a new issue and we'll investigate.
